services/workflows-service/src/errors.ts

```diff
@@ -25,24 +26,55 @@ export class SessionExpiredException extends common.UnauthorizedException {
   message!: string;
 }
 
-export class AjvValidationError extends common.BadRequestException {
+class DetailedValidationError {
+  @ApiProperty()
+  message!: string;
+
+  @ApiProperty()
+  path!: string;
+}
+
+export class ValidationError extends common.BadRequestException {
   @ApiProperty()
   statusCode!: number;
   @ApiProperty()
-  message!: string;
+  message: string = 'Validation error';
+
+  @ApiProperty({ type: DetailedValidationError })
+  errors!: { message: string; path: string }[];
 
-  constructor(
-    public error: ErrorObject<string, Record<string, any>, unknown>[] | null | undefined,
-  ) {
-    super();
+  constructor(errors: { message: string; path: string }[]) {
+    super(
+      {
+        statusCode: common.HttpStatus.BAD_REQUEST,
+        message: errors,
+        errors,
+      },
+      'Validation error',
+    );
+
+    this.errors = errors;
+  }
+
+  getErrors() {
+    return this.errors;
   }
 
-  serializeErrors() {
-    return this.error?.map(({ instancePath, message }) => {
-      return {
-        message: `${startCase(lowerCase(instancePath)).replace('/', '')} ${message}.`,
-        path: instancePath,
-      };
-    });
+  static fromAjvError(error: ErrorObject<string, Record<string, any>, unknown>[]) {
+    const errors = error.map(({ instancePath, message }) => ({
+      message: `${startCase(lowerCase(instancePath)).replace('/', '')} ${message}.`,
+      path: instancePath,
+    }));
+
+    return new ValidationError(errors);
+  }
+
+  static fromZodError(error: ZodError) {
+    const errors = error.errors.map(zodIssue => ({
+      message: zodIssue.message,
+      path: zodIssue.path.join('.'), // Backwards compatibility - Legacy code message excepts array
```

Comment:
Are we aware of what `path` can be? Wondering if joining with "." always outputs something like "user.details.email" and not something unexpected.

Reply:
changed and set . also for ajv